hal/src/main/native/sim/HAL.cpp

```diff
@@ -193,7 +201,68 @@ HAL_Bool HAL_GetBrownedOut(int32_t* status) {
   return false;  // Figure out if we need to detect a brownout condition
 }
 
+/* A return < 0 indicates an error that should stop the HAL.
+   0 is success, and > 0 is a non fatal error */
+typedef int halsim_init_func_t(void);
+
+int HAL_LoadExtraSimulation(const char* library) {
+  int rc = 1;  // It is expected and reasonable not to find an extra simulation
+#if defined(WIN32) || defined(_WIN32)
+  HMODULE handle = LoadLibrary(library);
+  if (!handle) {
+    std::string library_name = std::string(library) + ".dll";
+    handle = LoadLibrary(library_name.c_str());
+  }
+#else
+  void* handle = dlopen(library, RTLD_LAZY);
+  if (!handle) {
+    std::string library_name = "lib" + std::string(library) + ".so";
```

Comment:
This will have to be lib<library name>.dylib for OS X. Maybe move these platform-specific function calls and such to a single `#ifdef` block at the top of the file?  See [here](https://github.com/JacisNonsense/ToastCPP/blob/master/Toast-Core/include/toast/internal/loader.hpp) for an example

Reply:
Sure.  I can't test on Mac, so I'm hoping if I get it close I could ask you to test the last mile.

Reply:
Sure thing, I can test it on my mac whenever you're ready